src/constants.py

```diff
@@ -3,55 +3,26 @@
 """Constants."""
 from pathlib import Path
 
-SLURM_USER = "slurm"
-SLURM_GROUP = "slurm"
+_SNAP = Path("/snap/slurm/current")
+_SNAP_COMMON = Path("/var/snap/slurm/common")
 
-STATE_DIR = Path("/var/spool/slurmdbd")
+SLURM_USER = "root"
+SLURM_GROUP = "root"
+
+SLURMDBD_CONF_PATH = _SNAP_COMMON / "etc/slurm/slurmdbd.conf"
+SPOOL_DIR = _SNAP_COMMON / "var/lib/slurm/slurmdbd"
+SLURMDBD_DEFAULTS_FILE = _SNAP_COMMON / "etc/default/slurmdbd"
+JWT_RSA_PATH = SPOOL_DIR / "jwt_hs256.key"
+
+SLURM_ACCT_DB = "slurm_acct_db"
 
 CHARM_MAINTAINED_PARAMETERS = {
     "DbdPort": "6819",
     "AuthType": "auth/munge",
-    "AuthInfo": '"socket=/var/run/munge/munge.socket.2"',
+    "AuthInfo": f'"socket={_SNAP_COMMON}/run/munge/munged.socket.2"',
     "SlurmUser": SLURM_USER,
-    "PluginDir": "/usr/lib/x86_64-linux-gnu/slurm-wlm",
-    "PidFile": "/var/run/slurmdbd.pid",
-    "LogFile": "/var/log/slurm/slurmdbd.log",
+    "PluginDir": f"{_SNAP}/lib/slurm",
```

Comment:
Is this default configuration for `PluginDir` required here? Reason saying is that our [custom patch in the Slurm snap](https://github.com/charmed-hpc/slurm-snap/blob/main/patches/slurm/0001-set-default-plugin-and-config-location.patch) should already set the `PluginDir` location to `${SNAP}/lib/slurm`.

Reply:
I can test it without. I think it should still work